package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -0,0 +1,200 @@
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""
+Dihedral and Ramachandran analysis --- :mod:`MDAnalysis.analysis.dihedrals`
+===========================================================================
+
+:Author: Henry Mull
+:Year: 2018
+:Copyright: GNU Public License v2
+
+.. versionadded:: 0.18.1
+
+This module calculates the dihedral angles phi and psi in degrees for a given
+list of residues (or atoms corresponding to the residues). This can be done for
+selected frames or whole trajectories.
+
+A list of time steps that contain phi and psi angles for each residue is
+generated, and a basic Ramachandran plot can be generated using the attribute
+:meth:`Ramachandran.plot`. This plot is best used as a reference, but it also
+allows for user customization.
+
+
+See Also
+--------
+:mod:`MDAnalysis.lib.distances.calc_diherals`
+
+
+Example application
+-------------------
+This example will show how to calculate the phi and psi angles of adenylate
+kinase and generate a basic Ramachandran plot. The trajectory is included with
+the test data files::
+
+   import MDAnalysis as mda
+   from MDAnalysisTests.datafiles import GRO, XTC
+   u = mda.Universe(GRO, XTC)
+   r = u.select_atoms("protein")    # selection of residues
+
+   from MDAnalysis.analysis.dihedrals import Ramachandran
+   R = Ramachandran(r)
+   R.run()
+
+   import matplotlib.pyplot as plt
+   fig = plt.figure(figsize(5,5))
+   ax = fig.add_subplot(111)
+   ax.set_title("Ramachandran Plot (AdK)")
+   R.plot(ax=ax, color='k', marker='s')
+
+Alternatively, if you wanted to plot the data yourself, the angles themselves
+can be accessed using :meth:`Ramachandran.angles`::
+
+   fig = plt.figure(figsize(5, 5))
+   ax = fig.add_subplot(111)
+   ax.axis([-180,180,-180,180])
+   ax.axhline(0, color='k', lw=1)
+   ax.axvline(0, color='k', lw=1)
+   ax.set(xticks=range(-180,181,60), yticks=range(-180,181,60),
+          xlabel=r"$\phi$ (deg)", ylabel=r"$\psi$ (deg)")
+   for ts in R.angles:
+       ax.scatter(ts[:,0], ts[:,1], color='k', marker='s')
+
+Analysis Class
+--------------
+
+.. autoclass:: Ramachandran
+   :members:
+   :inherited-members:
+
+   .. attribute:: angles
```

Comment:
don't we want to have this as documentation in the class? @orbeckst is the information now rendered twice?